ledger-service/cli-opts/src/main/scala/cliopts/Logging.scala

```diff
@@ -7,29 +7,47 @@ import ch.qos.logback.classic.{Level => LogLevel}
 
 object Logging {
 
-  def reconfigure(clazz: Class[_], pathToLogbackFileInJarFile: String): Unit = {
+  sealed trait PathKind
+  object PathKind {
+    final case class OutsideOfJar(path: String) extends PathKind
+    final case class InsideOfJar(path: String) extends PathKind
+  }
+
+  def reconfigure(clazz: Class[_]): Unit = {
     // Try reconfiguring the library
     import ch.qos.logback.core.joran.spi.JoranException
     import ch.qos.logback.classic.LoggerContext
     import ch.qos.logback.classic.joran.JoranConfigurator
     import org.slf4j.LoggerFactory
     import scala.util.Using
-    Using.resource(clazz.getClassLoader.getResource(pathToLogbackFileInJarFile).openStream()) {
-      stream =>
+    import java.io.InputStream
+    import java.io.FileInputStream
+    val path: PathKind =
+      System.getProperty("logback.configurationFile") match {
+        case null => PathKind.InsideOfJar("logback.xml")
+        case path => PathKind.OutsideOfJar(path)
+      }
+    def reloadConfig(stream: => InputStream): Unit =
+      Using.resource(stream) { stream =>
         try {
           val context = LoggerFactory.getILoggerFactory.asInstanceOf[LoggerContext]
-          val configurator = new JoranConfigurator
+          val configurator = new JoranConfigurator()
           configurator.setContext(context)
           context.reset()
           configurator.doConfigure(stream)
         } catch {
           case je: JoranException =>
             // Fallback to System.err.println because the logger won't work in any way anymore.
-            System.err.println(s"reconfigured failed using url $pathToLogbackFileInJarFile: $je")
+            System.err.println(
+              s"reconfigured failed using url $path: $je"
+            )
             je.printStackTrace(System.err)
-        } finally {
-          stream.close()
-        }
+        } finally stream.close()
+      }
+    path match {
```

Comment:
seems a bit redundant to first build up `PathKind` above and then destruct it here. I think you can drop the whole thing and just inline the check here.

Reply:
I agree, no need to keep that type anymore.

Reply:
adressed with 2b830ae20e5665bbf1492159672d615dd789e788